recipes/imguizmo/all/conanfile.py

```diff
@@ -1,60 +1,67 @@
-import functools
-from conans import ConanFile, CMake, tools
+import os
+
+from conan import ConanFile
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+from conan.tools.files import copy, get
+
+required_conan_version = ">=1.53.0"
 
 
 class ImGuizmoConan(ConanFile):
     name = "imguizmo"
+    description = "Immediate mode 3D gizmo for scene editing and other controls based on Dear Imgui"
+    license = "MIT"
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://github.com/CedricGuillemet/ImGuizmo"
-    description = "Immediate mode 3D gizmo for scene editing and other controls based on Dear Imgui"
     topics = ("imgui", "3d", "graphics", "guizmo")
-    license = "MIT"
-    settings = "os", "arch", "compiler", "build_type"
-
-    exports_sources = ["CMakeLists.txt"]
-    generators = "cmake"
 
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
     options = {
         "shared": [True, False],
-         "fPIC": [True, False]
+        "fPIC": [True, False],
     }
     default_options = {
         "shared": False,
-        "fPIC": True
+        "fPIC": True,
     }
 
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def export_sources(self):
+        copy(self, "CMakeLists.txt", src=self.recipe_folder, dst=self.export_sources_folder)
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
+            self.options.rm_safe("fPIC")
 
-    def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def requirements(self):
-        self.requires("imgui/1.87")
+        self.requires("imgui/[<=1.89.3]", transitive_headers=True)
```

Comment:
Hi @valgur! Thanks a lot for the contribution. We still are trying version ranges in c3i only for CMake and OpenSSL, so this would be better set to a fixed version: https://github.com/conan-io/conan-center-index/blob/master/docs/adding_packages/dependencies.md#version-ranges